### ISSUE TITLE
HeavyIon addOn tests have incorrect module ordering

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#15807]
The new module/Path consistency check ( #15740 ) found the following problem in the two HI related addOn tests:
```
----- Begin Fatal Exception 10-Sep-2016 18:37:56 CEST-----------------------
An exception of category 'ScheduleExecutionFailure' occurred while
   [0] Calling beginJob
Exception Message:
Unrunnable schedule
Module run order problem found: 
CastorTowerReco after muons [path reconstruction_step], muons after hcalnoise [path reconstruction_step], hcalnoise after hiEvtPlane [path reconstruction_step], hiEvtPlane consumes CastorTowerReco
 Running in the threaded framework would lead to indeterminate results.
 Please change order of modules in mentioned Path(s) to avoid inconsistent module ordering.
----- End Fatal Exception -------------------------------------------------
```
Looking at the relavant part of the reconstruction_step Path does show the problem is real
```
+hiEvtPlane+hcalnoise+muons+CastorTowerReco
```

Where `hiEvtPlane` is 
```
>>> print process.hiEvtPlane.dumpPython()
cms.EDProducer("EvtPlaneProducer",
    CentBinCompression = cms.int32(5),
    FlatOrder = cms.int32(9),
    NumFlatBins = cms.int32(40),
    caloCentRef = cms.double(80.0),
    caloCentRefWidth = cms.double(5.0),
    caloTag = cms.InputTag("towerMaker"),
    castorTag = cms.InputTag("CastorTowerReco"),
    centralityBinTag = cms.InputTag("centralityBin","HFtowers"),
    centralityVariable = cms.string('HFtowers'),
    chi2 = cms.double(40.0),
    dzerr = cms.double(10.0),
    loadDB = cms.bool(False),
    maxet = cms.double(-1.0),
    maxpt = cms.double(3.0),
    maxvtx = cms.double(25.0),
    minet = cms.double(-1.0),
    minpt = cms.double(0.3),
    minvtx = cms.double(-25.0),
    nonDefaultGlauberModel = cms.string(''),
    trackTag = cms.InputTag("hiGeneralTracks"),
    vertexTag = cms.InputTag("hiSelectedVertex")
)
```
It is impossible for `hiEvtPlane` to get the `CastorTowerReco` it does need.

The upcoming threaded framework update no longer tolerates this behavior and requires that all Paths and consumes create a consistent schedule.

Please either move `hiEvtPlane` later in the `reconstruction_step` or change `hiEvtPlane. castorTag` to tell the code to get its value from somewhere else.

**Comment by Dr15Jones** [comment]
assign reconstruction

**Comment by Dr15Jones** [comment]
assign hlt

**Comment by Dr15Jones** [comment]
assign core

**Comment by slava77** [comment]
are these workflows unscheduled already? Does the issue remain in unscheduled mode?

We have quite a bit of "legacy" that was never converted to unscheduled, while the central workflows for the equivalents are unscheduled.

**Comment by Dr15Jones** [comment]
These workflows are not unscheduled. See 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15740/15080/addOnTests/hlt_data_HIon/RelVal_HLT_RAW2DIGI_L1Reco_RECO.py

However, since this is a combined "HLTRECO" job, are those rated to work as unscheduled?

**Comment by Martin-Grunewald** [comment]
Since HLT is not unscheduled, I do not think HLT+RECO could be unscheduled...

**Comment by Martin-Grunewald** [comment]
In any case I would assume a "@skipCurrentProcess" is simply missing...?

**Comment by Dr15Jones** [comment]
The data the module wants is there, it is just generated by a module that is run just a couple of slots later in the same Path.

**Comment by Dr15Jones** [comment]
'@skipCurrentProcess' would make it behave the same as it does now, but I think I found a true bug which was being hidden by modules not reporting when they can't get the data they need.

**Comment by Dr15Jones** [comment]
I think the problem could be fixed by changing line 56
http://cmslxr.fnal.gov/source/Configuration/StandardSequences/python/ReconstructionHeavyIons_cff.py?v=CMSSW_8_1_X_2016-09-11-2300#0056

to be
```
reconstruct_PbPb = cms.Sequence(localReco*CastorFullReco*globalRecoPbPb)
```
although it is unclear since `muonRecoHighLevelPbPb` is what is at the end of `globalRecoPbPb` but from the printout from the failed job it is `muons`.

**Comment by Dr15Jones** [comment]
Fixed by #15815
